substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeBootImage.java

```diff
@@ -562,7 +568,8 @@ private static void markDataRelocationSite(final ProgbitsSectionImpl sectionImpl
     }
 
     private void markDataRelocationSiteFromText(RelocatableBuffer buffer, final ProgbitsSectionImpl sectionImpl, final int offset, final Info info, final Map<Object, ObjectInfo> objectMap) {
-        assert ((info.getRelocationSize() == 4) || (info.getRelocationSize() == 8)) : "Data relocation size should be 4 or 8 bytes.";
+// assert ((info.getRelocationSize() == 4) || (info.getRelocationSize() == 8)) : "Data relocation
```

Comment:
We have to double check on which side we are doing wrong and either fix the generating code or this assertion.

Reply:
Our relocations aren’t byte-aligned sizes so it’s not valid for aarch64